File: Array/liuchang/day36/day36.java
public class day36 {


    public static int count4One(int num) {
        int tmp, count = 0;
        while (num != 0) {
            tmp = num & 1;
            if (tmp == 1)
                count++;
            num = num >> 1;
        }
        return count;
    }

    public static void main(String[] args) {
        System.out.println( count4One(15));
    }
}

File: Array/liuchang/day36/day36.kt
package net.study.kotlintest

/**
 * @author       刘畅
 * @createdate   2020/7/7
 * @describe
 */
class day36 {

    fun count4One(num: Int): Int {
        var tmp = 0;
        var count = 0;
        var number = num;
        while (number != 0) {
            tmp = (number.and(1))
            if (tmp == 1)
                count++;
            number = number.shr(1) ;
        }
        return count;
    }

    fun main(args:Array<String>){

        System.out.println("=====" + count4One(15));
    }
}